src/abstractdataframe/selection.jl

```diff
@@ -612,7 +619,7 @@ See [`select`](@ref) for examples.
 select!(df::DataFrame, @nospecialize(args...); renamecols::Bool=true) =
     _replace_columns!(df, select(df, args..., copycols=false, renamecols=renamecols))
 
-function select!(arg::Base.Callable, df::AbstractDataFrame; renamecols::Bool=true)
+function select!(@nospecialize(arg::Base.Callable), df::AbstractDataFrame; renamecols::Bool=true)
```

Comment:
Likewise, I don't think this should make any difference if the input is a `Function`. Specialization may still happen e.g. for `Colon` but we throw an error so we probably don't care, right?

Reply:
It seems it is not necessary as you commented in this case, but does it hurt to add `@nospecialize` (frankly - I automatically identified all methods that take a function or a `NamedTuple` and annotated them as these the most offending types that cause latency).

The question is if adding `@nospecialize` is a problem (if not then at least it is a clear signal what we intend).

Reply:
Yeah not a big deal. I wanted to make sure we understand what's going on.